flink-python/pyflink/datastream/data_stream.py

```diff
@@ -434,6 +465,16 @@ def flat_map(self, func: Union[Callable, FlatMapFunction], type_info: TypeInform
             -> 'DataStream':
         return self._values().flat_map(func, type_info)
 
+    def filter(self, func: Union[Callable, FilterFunction]) -> 'DataStream':
+        return self._values().filter(func)
+
+    def add_sink(self, sink_func: SinkFunction) -> 'DataStreamSink':
+        return self._values().add_sink(sink_func)
+
+    def key_by(self, key_selector: Union[Callable, KeySelector],
+               key_type_info: TypeInformation = None) -> 'KeyedStream':
+        return self._values().key_by(key_selector, key_type_info)
```

Comment:
There are side effects for this implementation. If we perform multi key_by, it will also introduce multi maps in the stream graph. This should be avoided.